code/modules/admin/create_poll.dm

```diff
@@ -1,162 +1,202 @@
+	var/polltype = href_list["createpoll"]
+	if(polltype != POLLTYPE_OPTION && polltype != POLLTYPE_TEXT && polltype != POLLTYPE_RATING && polltype != POLLTYPE_MULTI)
+		create_poll_window("<font color='red'>Invalid poll type</font>")
 		return
-	if(query_validate_time.NextRow())
-		endtime = query_validate_time.item[1]
-		if(!endtime)
-			to_chat(src, "Datetime entered is invalid.")
+	var/choice_amount = 0
+	if(polltype == POLLTYPE_MULTI)
+		choice_amount = text2num(href_list["choiceamount"])
+		if(!isnum(choice_amount) || choice_amount < 1)
+			create_poll_window("<font color='red'>Invalid choice amount. Must be at least 1.</font>")
 			return
-	var/DBQuery/query_time_later = dbcon.NewQuery("SELECT TIMESTAMP('[endtime]') < NOW()")
-	if(!query_time_later.Execute())
-		var/err = query_time_later.ErrorMsg()
-		log_game("SQL ERROR comparing endtime to NOW(). Error : \[[err]\]\n")
+	var/poll_len = text2num(href_list["createpollduration"])
+	if(!isnum(poll_len) || poll_len < 1)
+		create_poll_window("<font color='red'>Invalid poll duration. Must be at least 1.</font>")
 		return
-	if(query_time_later.NextRow())
-		var/checklate = text2num(query_time_later.item[1])
-		if(checklate)
-			to_chat(src, "Datetime entered is not later than current server time.")
-			return
-	var/adminonly
-	switch(alert("Admin only poll?",,"Yes","No","Cancel"))
-		if("Yes")
-			adminonly = 1
-		if("No")
-			adminonly = 0
-		else
-			return
+	var/adminonly = text2num(href_list["createpolladminonly"]) ? 1 : 0
 	var/sql_ckey = sanitizeSQL(ckey)
-	var/question = input("Write your question","Question") as message|null
+	var/question = href_list["createpollquestion"]
 	if(!question)
+		create_poll_window("<font color='red'>Question cannot be blank.</font>")
 		return
 	question = sanitizeSQL(question)
-	var/DBQuery/query_polladd_question = dbcon.NewQuery("INSERT INTO [format_table_name("poll_question")] (polltype, starttime, endtime, question, adminonly, multiplechoiceoptions, createdby_ckey, createdby_ip) VALUES ('[polltype]', '[starttime]', '[endtime]', '[question]', '[adminonly]', '[choice_amount]', '[sql_ckey]', '[address]')")
-	if(!query_polladd_question.Execute())
-		var/err = query_polladd_question.ErrorMsg()
+	
+	var/starttime
+	var/endtime
+	var/DBQuery/query = dbcon.NewQuery("SELECT Now() AS starttime, ADDDATE(Now(), INTERVAL [poll_len] DAY) AS endtime")
+	query.Execute()
+	while(query.NextRow())
+		starttime = query.item[1]
+		endtime = query.item[2]
+	
+	var/pollquery = "INSERT INTO [format_table_name("poll_question")] (polltype, starttime, endtime, question, adminonly, multiplechoiceoptions, createdby_ckey, createdby_ip) VALUES ('[polltype]', '[starttime]', '[endtime]', '[question]', '[adminonly]', '[choice_amount]', '[sql_ckey]', '[address]')"
+	var/idquery = "SELECT id FROM [format_table_name("poll_question")] WHERE question = '[question]' AND starttime = '[starttime]' AND endtime = '[endtime]' AND createdby_ckey = '[sql_ckey]' AND createdby_ip = '[address]'"
```

Comment:
It's sanitized on https://github.com/ParadiseSS13/Paradise/pull/6596/files#diff-463efed0062b39a85edb29fa9f52ef6cL90 though...